raft.go

```diff
@@ -753,6 +798,19 @@ func (r *Raft) processLog(l *Log, future *logFuture) {
 // processRPC is called to handle an incoming RPC request. This must only be
 // called from the main thread.
 func (r *Raft) processRPC(rpc RPC) {
+	if v, ok := rpc.Command.(WithVersionInfo); ok {
+		if !r.isVersionCompatible(v.GetVersionInfo()) {
+			r.logger.Printf("[ERR] raft: Ignoring unsupported %#v for command: %#v", v, rpc.Command)
+			rpc.Respond(nil, ErrUnsupportedProtocol)
+			return
+		}
+
+	} else {
+		r.logger.Printf("[ERR] raft: Ignoring un-versioned command: %#v", rpc.Command)
```

Comment:
How does this work? Don't we need to listen to unversioned commands until we've reached some version ourselves?